### ISSUE TITLE
Compression of messages

### ISSUE THREAD

**Issue by Mitiko** [Azer0s/Olymp#13]
**Is your feature request related to a problem? Please describe.**
Compression of messages will both save time and bandwidth. Using no compression will result in slower response times and bigger network bills (if used in production)

**Describe the solution you'd like**
Compress messages both in the context of the message itself and in the context of previous messages transmitted. The out-of-the-box defaults will be something fast like gzip or something with LZ78 like LZW. Ability to implement your own compression in order to optimize performance is also gonna be pretty cool. The default settings will be tuned for an example application with an example dataset. The end user should also be able to have control over compression options (preferably based on a messages lower limit size, i.e. only applied on messages > 100mb)  

**Describe alternatives you've considered**
Not using compression

**Additional context**
To qualify to give a solution to this issue, you must have watched all seasons of Silicon Valley and CompressorHead by Colt McAnlis.

**Comment by Azer0s** [comment]
> To qualify to give a solution to this issue, you must have watched all seasons of Silicon Valley and CompressorHead by Colt McAnlis

lol 😂

**Comment by Azer0s** [comment]
Issue approved without remarks

**Comment by Azer0s** [comment]
@alexmandt sounds like a good issue for you. Can I assign you?

**Comment by alexmandt** [comment]
If the issue is not hight priority, then yeah, I can do it. I can currently only work on programming stuff in the evening because of other stuff that I have to do right now.

**Comment by alexmandt** [comment]
So stuff that needs to get done quickly is not something that I can work on.

**Comment by Mitiko** [comment]
@alexmandt There's no quick in 'quality code'. I could help of you need any.

@Azer0s Can we please do a custom solution?

**Comment by alexmandt** [comment]
@Mitiko  No. I mean I cannot work on the project the entire day. I only have 1 to 2 Hours everyday to work on this project. So if it is high priority, then I am not the best fit.

**Comment by Azer0s** [comment]
@alexmandt This is not high priority. We just need a smart person, like you, to work on a good solution. I don't know the first thing about compression on network communication.

@Mitiko What? You mean like middle out 😆

**Comment by alexmandt** [comment]
Okay. Then assign me.

**Comment by Mitiko** [comment]
@alexmandt If you want to work on something high priority, help me with the child node and child node worker communication.

@Azer0s The show's idea is cool, but I'm serious. Compression gets me high like no other CS concept. Markov is my middle name, along with Lempel, Ziv, Huffman, Dynamic Arithmetic (which is better than Huffman btw)...

**Comment by alexmandt** [comment]
@Mitiko I can help out with high priority stuff but I only have limited time available, so please don't expect much from me.
